sympy/polys/polytools.py

```diff
@@ -6218,8 +6218,20 @@ def sqf_list(f, *gens, **args):
     (2, [(x + 1, 2), (x + 2, 3)])
 
     """
-    return _generic_factor_list(f, gens, args, method='sqf')
-
+    options.allowed_flags(args, ['frac', 'polys'])
+    opt = options.build_options(gens, args)
+    f = sympify(f)
+    if isinstance(f, Expr) and not f.is_Relational:
+        numer, denom = together(f).as_numer_denom()
+        numer = Poly(numer, *gens, **args)
```

Comment:
It is probably possible to use `_poly_from_expr(f, opt)`. It may raise an error. Then it should be checked that the result is univariate.

Reply:
The system hangs when it runs `assert sqf_list(f) == (1, [(x + 1, 40000000000)])` as `_poly_from_expr(f, opt)` is returning the expanded form of polynomial. Is there something to stop this?
Also, what needs to be done for `sqf_list(1)`, should we provide the output or raise an error asking for generator?